src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
The cURL extension will attempt to reuse connections automatically and for the PB lib I am using persistent sockets, so if they are using PHP on a webserver, there will be some performance boosts there. On the CLI side, I am pretty sure that there is no way to reuse connections unless I create a local proxy with another language that can stay open listening for requests from PHP. 

Even with that said, the reuse by cURL and sockets is only available for the same process of PHP, e.g. FastCGI runs a master process and several child processes, the reuse will work only for that child process.